src/llm/llmnoderesources.hpp

```diff
@@ -37,6 +37,7 @@
 
 #include "../logging.hpp"
 #include "../stringutils.hpp"
+#include "llm_executor.hpp"
```

Comment:
Why is it needed here now?

Reply:
You are right, it is not needed, we can forward declare. After doing so I could even decrease number of expected errors in ci/cpp_clean.sh